react-router/index.ts

```diff
@@ -0,0 +1,18 @@
+import { match, matchPath } from 'react-router';
+
+export const createMatchSelector = (path: string) => {
+   let lastPathname: string | null = null;
+   let lastMatch: match<{}> | null = null;
+   return (state: any) => {
+      const { pathname } = state.router.location ?? {};
+      if (pathname === lastPathname) {
```

Comment:
Should use selectRouterState

Reply:
> Hi, thanks for this PR!
> 
> Some points I'd like to care about:
> 
> 1. state.router.location -> should use "selectRouterState" function as createReduxHistoryContext options.
> 2. Why is in the react-router folder?

Hi, thanks for response!
Usage of selectRouterState function - done
Separated folder - it's a specially react-router function that use "matchPath" from react-router

Reply:
Does it apply only to react router v5?

Reply:
**v5** and **v6**, used function exists in both
[](https://v5.reactrouter.com/web/api/matchPath)
[](https://reactrouter.com/docs/en/v6/api#matchpath)